dotnet/src/SemanticKernel.Abstractions/Functions/KernelFunction.cs

```diff
@@ -144,14 +168,22 @@ internal KernelFunction(string name, string description, IEnumerable<AIRequestSe
 
         logger.LogInformation("Function streaming invoking.");
 
-        cancellationToken.ThrowIfCancellationRequested();
+        // Invoke pre hook, and stop if skipping requested.
```

Comment:
Thanks for adding this, I added in my current UT branch, will use yours when merging.